org.hl7.fhir.validation/src/main/java/org/hl7/fhir/validation/cli/services/ValidationService.java

```diff
@@ -253,6 +253,37 @@ public void transform(CliContext cliContext, ValidationEngine validator) throws
     }
   }
 
+  public void compile(CliContext cliContext, ValidationEngine validator) throws Exception {
+    if (cliContext.getSources().size() > 0)
+      throw new Exception("Cannot specify sources when compling transform (found " + cliContext.getSources() + ")");
+//    if (cliContext.getTxServer() == null)
+//      throw new Exception("Must provide a terminology server when compiling a transform");
+    if (cliContext.getMap() == null)
+      throw new Exception("Must provide a map when compiling a transform");
+    try {
+      List<StructureDefinition> structures = validator.getContext().allStructures();
+      for (StructureDefinition sd : structures) {
+        if (!sd.hasSnapshot()) {
+          if (sd.getKind() != null && sd.getKind() == StructureDefinitionKind.LOGICAL) {
+            validator.getContext().generateSnapshot(sd, true);
+          } else {
+            validator.getContext().generateSnapshot(sd, false);
+          }
+        }
+      }
+      validator.setMapLog(cliContext.getMapLog());
+      cliContext.getMap();
+      StructureMap map = validator.compile(cliContext.getMap());
+      System.out.println(" ...success");
```

Comment:
Please don't use System.out ... it clutters the terminal

Reply:
I was just being consistent with existing behavior - which already uses System.out elsewhere.

Reply:
@jvitrifork is correct, but @lmckenzi is being consistent. Unfortunately System.out is used throughout to provide output for the command line interface.